base_layer/core/src/validation/mocks.rs

```diff
@@ -115,10 +115,10 @@ impl<B: BlockchainBackend> HeaderChainLinkedValidator<B> for MockValidator {
         _: Option<Difficulty>,
     ) -> Result<AchievedTargetDifficulty, ValidationError> {
         if self.is_valid.load(Ordering::SeqCst) {
-            let achieved = sha3x_difficulty(header);
+            let achieved = sha3x_difficulty(header)?;
 
             let achieved_target =
-                AchievedTargetDifficulty::try_construct(PowAlgorithm::Sha3, achieved - Difficulty::from(1), achieved)
+                AchievedTargetDifficulty::try_construct(PowAlgorithm::Sha3, achieved, achieved.checked_add(1).unwrap())
```

Comment:
These 2 lines are not equal:
We had` (sha3, achieved -1, achieved)`
Now we have: `(sha3, achieved, achieved + 1)`

Reply:
The original line in the test now panics because it results in a difficulty of zero as these tests use localnet with min and max difficulties set to 1.